ledger/store/src/helpers/rocksdb/internal/map.rs

```diff
@@ -109,55 +111,55 @@ impl<
     ///
     fn atomic_checkpoint(&self) {
         // Push the current length of the atomic batch to the checkpoint stack.
-        self.checkpoint.lock().push(self.atomic_batch.lock().len());
+        self.checkpoints.lock().push(self.atomic_batch.lock().len());
     }
 
     ///
     /// Removes the latest atomic checkpoint.
     ///
     fn clear_latest_checkpoint(&self) {
         // Removes the latest checkpoint.
-        let _ = self.checkpoint.lock().pop();
+        let _ = self.checkpoints.lock().pop();
     }
 
     ///
     /// Removes all pending operations to the last `atomic_checkpoint`
     /// (or to `start_atomic` if no checkpoints have been created).
     ///
     fn atomic_rewind(&self) {
-        // Acquire the write lock on the atomic batch.
-        let mut atomic_batch = self.atomic_batch.lock();
-
         // Retrieve the last checkpoint.
-        let checkpoint = self.checkpoint.lock().pop().unwrap_or(0);
-
+        let checkpoint = self.checkpoints.lock().pop().unwrap_or(0);
         // Remove all operations after the checkpoint.
-        atomic_batch.truncate(checkpoint);
+        self.atomic_batch.lock().truncate(checkpoint);
     }
 
     ///
     /// Aborts the current atomic operation.
     ///
     fn abort_atomic(&self) {
         // Clear the atomic batch.
-        *self.atomic_batch.lock() = Default::default();
+        self.atomic_batch.lock().clear();
         // Clear the checkpoint stack.
-        *self.checkpoint.lock() = Default::default();
+        self.checkpoints.lock().clear();
         // Set the atomic batch flag to `false`.
         self.batch_in_progress.store(false, Ordering::SeqCst);
+        // Clear the database-wide atomic batch.
+        self.database.atomic_batch.lock().clear();
+        // Reset the atomic batch depth.
+        self.database.atomic_depth.store(0, Ordering::SeqCst);
```

Comment:
I believe setting the `atomic_depth` to 0 here is correct, but is there ever a case where we should be decrementing by 1, and then ensuring on the last `abort_atomic` call, that we hit 0? Meaning we've properly aborted all maps?

Reply:
`abort_atomic` used to "bubble up" in the past, but this has recently changed; now, instead of being called in case of errors, it is only called in `atomic_finalize`, after all the other operations.